pkg/minikube/bootstrapper/kubeadm/kubeadm.go

```diff
@@ -638,124 +583,41 @@ func (k *Bootstrapper) restartControlPlane(cfg config.ClusterConfig) error {
-		return errors.Wrap(err, "apiserver health")
-	}
-
-	// because reboots clear /etc/cni
-	if err := k.applyCNI(cfg); err != nil {
-		return errors.Wrap(err, "apply cni")
-	}
-
-	if err := kverify.WaitForSystemPods(cr, k, cfg, k.c, client, time.Now(), kconst.DefaultControlPlaneTimeout); err != nil {
-		return errors.Wrap(err, "system pods")
-	}
-
-	if err := kverify.NodePressure(client); err != nil {
-		adviseNodePressure(err, cfg.Name, cfg.Driver)
-	}
-
-	// This can fail during upgrades if the old pods have not shut down yet
-	addonPhase := func() error {
-		_, err := k.c.RunCmd(exec.Command("/bin/bash", "-c", fmt.Sprintf("%s phase addon all --config %s", baseCmd, conf)))
-		return err
+		klog.Infof("needs reconfigure: apiserver error: %v", err)
+		return true
 	}
-	if err = retry.Expo(addonPhase, 100*time.Microsecond, 30*time.Second); err != nil {
-		klog.Warningf("addon install failed, wil retry: %v", err)
-		return errors.Wrap(err, "addons")
+	if st != state.Running {
+		klog.Infof("needs reconfigure: apiserver in state %s", st)
+		return true
 	}
 
-	// must be called after applyCNI and `kubeadm phase addon all` (ie, coredns redeploy)
-	if cfg.VerifyComponents[kverify.ExtraKey] {
-		// after kubelet is restarted (with 'kubeadm init phase kubelet-start' above),
-		// it appears as to be immediately Ready as well as all kube-system pods (last observed state),
-		// then (after ~10sec) it realises it has some changes to apply, implying also pods restarts,
-		// and by that time we would exit completely, so we wait until kubelet begins restarting pods
-		klog.Info("waiting for restarted kubelet to initialise ...")
-		start := time.Now()
-		wait := func() error {
-			pods, err := client.CoreV1().Pods(meta.NamespaceSystem).List(context.Background(), meta.ListOptions{LabelSelector: "tier=control-plane"})
-			if err != nil {
-				return err
-			}
-			for _, pod := range pods.Items {
-				if ready, _ := kverify.IsPodReady(&pod); !ready {
-					return nil
-				}
-			}
-			return fmt.Errorf("kubelet not initialised")
-		}
-		_ = retry.Expo(wait, 250*time.Millisecond, 1*time.Minute)
-		klog.Infof("kubelet initialised")
-		klog.Infof("duration metric: took %s waiting for restarted kubelet to initialise ...", time.Since(start))
-
-		if err := kverify.WaitExtra(client, kverify.CorePodsLabels, kconst.DefaultControlPlaneTimeout); err != nil {
-			return errors.Wrap(err, "extra")
-		}
+	if err := kverify.ExpectAppsRunning(client, kverify.AppsRunningList); err != nil {
+		klog.Infof("needs reconfigure: %v", err)
+		return true
 	}
 
-	if err := bsutil.AdjustResourceLimits(k.c); err != nil {
-		klog.Warningf("unable to adjust resource limits: %v", err)
+	if err := kverify.APIServerVersionMatch(client, cfg.KubernetesConfig.KubernetesVersion); err != nil {
+		klog.Infof("needs reconfigure: %v", err)
+		return true
 	}
 
-	return nil
+	// DANGER: This log message is hard-coded in an integration test!
```

Comment:
I wonder if we can use a exported const that the integration test can see?

Reply:
Moved to the `constants` package